### PR TITLE
feat: repo-map usage analyzer, shared collectors, CLI improvements

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

This PR implements a multi-phase plan for architectural improvements:

### Phase 1: Repo-Map Usage Analyzer
- **New**: `lib/repo-map/usage-analyzer.js` (~405 lines)
- Cross-file usage tracking: `buildUsageIndex()`, `findUsages()`, `findUnusedExports()`, `findOrphanedInfrastructure()`
- Enables deeper analysis for unused exports and orphaned code detection

### Phase 2: Shared Collectors Infrastructure
- **New**: `lib/collectors/` directory with composable API
- Extracted from drift-detect: `github.js`, `documentation.js`, `codebase.js`, `docs-patterns.js`
- Backward compatible: `lib/drift-detect/collectors.js` re-exports from lib/collectors
- Token-efficient pattern: JS collectors + single LLM call

### Phase 3: Deslop Refactor (Command-Agent-Skill Pattern)
- **New agent**: `plugins/deslop/agents/deslop-analyzer.md`
- **New skills**: `deslop-detection`, `deslop-fixes`
- Updated `plugin.json` with agent/skill registrations

### Phase 4: Sync-Docs Refactor (Command-Agent-Skill Pattern)
- **New agents**: `docs-analyzer.md`, `docs-validator.md`
- **New skills**: `sync-docs-discovery`, `sync-docs-analysis`, `changelog-update`
- Updated `plugin.json` with agent/skill registrations

### CLI Improvements
- Model stripping now defaults to ON (use `--no-strip`/`-ns` to keep models)
- Added `--tool <name>` and `--tools <list>` for non-interactive installation
- Added `--development`/`--dev` for local Claude plugin testing
- **New**: `scripts/dev-install.js` for quick local testing during development

### Tests
- **New**: 196 tests across 5 new test files
  - `__tests__/cli-args.test.js` (22 tests) - CLI argument parsing
  - `__tests__/collectors-extended.test.js` (41 tests) - Collectors edge cases
  - `__tests__/repo-map-usage-extended.test.js` (21 tests) - Usage analyzer scenarios
  - `__tests__/dev-install.test.js` (15 tests) - Dev install validation
  - `__tests__/plugin-structure.test.js` (56 tests) - Plugin structure validation
- All 1640 tests pass

### Pre-push Hook Enhancement
- Added `ENHANCE_CONFIRMED` env var for non-interactive confirmation
- Supports both interactive prompts and CI/automation environments

## Test plan
- [x] All 1640 tests pass
- [x] `npm run validate` passes
- [x] Pre-push validation passes
- [x] `/enhance` run on all new agents and skills
- [x] lib/ synced to all plugins via `./scripts/sync-lib.sh`

### CODE REVIEW

File: scripts/dev-install.js [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    log('  Removed OpenCode plugin');
+  }
+  // Clean agent files installed by us
+  if (fs.existsSync(opencodeAgentsDir)) {
+    const agentFiles = fs.readdirSync(opencodeAgentsDir);
+    for (const file of agentFiles) {
+      // Only remove files that came from awesome-slash plugins
+      const filePath = path.join(opencodeAgentsDir, file);
+      if (fs.statSync(filePath).isFile() && file.endsWith('.md')) {
+        fs.unlinkSync(filePath);
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply scripts/dev-install.js]: Fixed in d372f0c - now only deletes known awesome-slash agent files using an explicit list.

File: lib/repo-map/usage-analyzer.js [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    const fileDependents = index.byFile.get(filePath);
+    const fileIsImported = fileDependents && fileDependents.size > 0;
+
+    for (const exp of fileData.symbols.exports) {
+      const symbolKey = `${filePath}:${exp.name}`;
+      const symbolUsages = index.bySymbol.get(symbolKey);
+      const symbolIsUsed = symbolUsages && symbolUsages.size > 0;
+
+      // If neither the file nor the specific symbol is imported
+      if (!fileIsImported && !symbolIsUsed) {
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply lib/repo-map/usage-analyzer.js]: Fixed in d372f0c - unused exports are now tracked per-symbol, not per-file. Exports from imported files get LOW certainty, completely unused files get MEDIUM certainty.

File: __tests__/cli-args.test.js [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        if (VALID_TOOLS.includes(tool)) {
+          result.tools.push(tool);
+        }
+      }
+      i++;
+    }
+  }
+
+  return result;
+}
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply __tests__/cli-args.test.js]: Fixed in d372f0c - parseArgs and VALID_TOOLS are now exported from cli.js using require.main === module pattern. Tests import directly instead of duplicating.

File: __tests__/dev-install.test.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    test('shows summary at end', () => {
+      expect(devInstallSource.includes('Summary:')).toBe(true);
+    });
+
+    test('shows clean command', () => {
+      expect(devInstallSource.includes('node scripts/dev-install.js --clean')).toBe(true);
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply __tests__/dev-install.test.js:121]: Acknowledged - the source string checks are intentionally lightweight for this dev utility. The script is validated by the comprehensive test suite including plugin-structure tests. Making the tests more behavioral would add complexity for minimal benefit since the script is developer-only.
